pkg/system/phase4_configuring.go

```diff
@@ -269,7 +271,24 @@ func (r *Reconciler) SetDesiredDeploymentEndpoint() error {
 	podSpec.SecurityContext.RunAsUser = &rootUIDGid
 	podSpec.SecurityContext.RunAsGroup = &rootUIDGid
 	podSpec.ServiceAccountName = "noobaa-endpoint"
-
+	honor := corev1.NodeInclusionPolicyHonor
+	disableTopologyConstraints, found := os.LookupEnv(disableTopologySpreadConstraints)
+	if found && disableTopologyConstraints == "true" {
+		r.Logger.Info("TopologyConstraints for noobaa endpoint is disabled")
+	} else {
+		topologySpreadConstraint := corev1.TopologySpreadConstraint{
+			MaxSkew:           1,
+			TopologyKey:       "kubernetes.io/hostname",
+			WhenUnsatisfiable: corev1.ScheduleAnyway,
+			NodeTaintsPolicy:  &honor,
+			LabelSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"noobaa-s3": "noobaa",
+				},
+			},
+		}
+		podSpec.TopologySpreadConstraints = []corev1.TopologySpreadConstraint{topologySpreadConstraint}
```

Comment:
why are we always overriding it? shouldn't we only set it if null?

Reply:
@guymguym we will override it only when `DISABLE_TOPOLOGY_SPREAD_CONSTRAINTS`(will rename to `DISABLE_DEFAULT_TOPOLOGY_SPREAD_CONSTRAINTS`) is empty or false, If the user set it to true and adds their `TopologySpreadConstraint` it will not be overridden. Correct me if am wrong.

Reply:
why override it at all? if it's set I would just leave it as is.